packages/debug/src/browser/debug-prefix-configuration.ts

```diff
@@ -148,9 +155,9 @@ export class DebugPrefixConfiguration implements CommandContribution, CommandHan
      *
      * @param configuration the `DebugSessionOptions`.
      */
-    protected runConfiguration(configuration: DebugSessionOptions): void {
+    protected async runConfiguration(configuration: DebugSessionOptions): Promise<void> {
         this.debugConfigurationManager.current = { ...configuration };
-        this.commandRegistry.executeCommand(DebugCommands.START.id);
+        await this.commandRegistry.executeCommand(DebugCommands.START.id);
```

Comment:
It looks like a more direct strategy might be to modify this code. The `DebugCommands.Start` command accepts an optional argument `DebugSessionOptions` to be run, and falls back to `debugConfigurationManager.current` if none is supplied. Given that, I believe that passing `configuration` as a second argument to `executeCommand` should successfully run the configuration without modifying the `current` field of the `debugConfigurationManager`.

Reply:
Good catch Colin!
Since there are two different behaviours when exectuing 'dynamic' vs 'non dynamic' configurations, 
I created a new function for 'dynamic', I could inline this call rather than creating a new function although I think it's clearer this way.
Comments are very welcome !